edgexfoundry/device-rest-go
# refactor!: Changed configuration and simple-devices files format to YAML

BREAKING CHANGE: Configuration and sample-devices files are now in YAML format

closes issue #280







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rest-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions
/device-rest-go$ make clean
/device-rest-go$ make build
/device-rest-go$ make docker
/device-rest-go$ make test without error
edgex-compose/compose-builder$ make pull
edgex-compose/compose-builder$ make run no-secty
device-rest-go/cmd$ ./device-rest -cp -o
verify that below message in the log without error:

```
level=INFO ts=2023-03-28T18:45:53.995919449Z app=device-rest source=config.go:565 msg="Loading configuration file from ./res/configuration.yaml"
...
level=INFO ts=2023-03-28T18:45:54.077009243Z app=device-rest source=devices.go:51 msg="Loading pre-defined devices from /home/v/source/repos/device-rest-go/cmd/res/devices(1 files found)"
level=INFO ts=2023-03-28T18:45:54.077807678Z app=device-rest source=devices.go:92 msg="Device sample-json not found in Metadata, adding it ..."
level=INFO ts=2023-03-28T18:45:54.077914571Z app=device-rest source=devices.go:92 msg="Device sample-image not found in Metadata, adding it ..."
level=INFO ts=2023-03-28T18:45:54.077963557Z app=device-rest source=devices.go:92 msg="Device sample-numeric not found in Metadata, adding it ..."
level=INFO ts=2023-03-28T18:45:54.078002496Z app=device-rest source=devices.go:92 msg="Device 2way-rest-device not found in Metadata, adding it ..."
```

## New Dependency Instructions (If applicable)
